openmcyclus/depletion.py

```diff
@@ -0,0 +1,155 @@
+            power output of the reactor, assumed in MWe.
+        '''
+        self.path = path
+        self.prototype = prototype
+        self.chain_file = chain_file
+        self.timesteps = timesteps
+        self.power = power
+
+    def read_model(self):
+        '''
+        Read the .xml files in the defined path to create an OpenMC
+        model in the python API. The files are assumed to be named
+        geometry.xml, materials.xml, settings.xml, and (optionally)
+        tallies.xml
+
+        Returns:
+        ---------
+        model: openmc.model.model object
+            OpenMC model of reactor geometry.
+        '''
+        model = openmc.Model.from_xml(
+            geometry=str(
+                self.path +
+                "geometry.xml"),
+            materials=str(
+                self.path +
+                "materials.xml"),
+            settings=str(
+                self.path +
+                "materials.xml"))
+        return model
+
+    def read_microxs(self):
+        '''
+        Reads .csv file with microscopic cross sections. The
+        csv file is assumed to be named "micro_xs.csv". This will need to
+            be relative to Cyclus input file location
+
+        Parameters:
+        -----------
+
+        Returns:
+        --------
+        microxs: object
+            microscopic cross section data
+        '''
+        microxs = od.MicroXS.from_csv(str(self.path + "micro_xs.csv"))
+        return microxs
+
+    def run_depletion(self):
+        '''
+        Run the IndependentOperator class in OpenMC to perform
+        stand-alone depletion.
+
+        Parameters:
+        -----------
+
+        Outputs:
+        --------
+        depletion_results.h5: database
+            HDF5 data base with the results of the depletion simulation
+        '''
+        model = self.read_model()
+        micro_xs = self.read_microxs()
+        ind_op = od.IndependentOperator(model.materials, micro_xs,
+                                        str(self.path + self.chain_file))
+        integrator = od.PredictorIntegrator(ind_op, np.ones(
+            self.timesteps), power=self.power, timestep_units='d')
+        integrator.integrate()
+        os.system(
+            'mv ./depletion_results.h5 ' +
+            self.path +
+            "depletion_results.h5")
+        return
+
+    def create_recipe(self):
+        '''
+        Converts the depleted material compositions to an XML file readable
+        by cyclus.
+
+        Parameters:
+        -----------
+
+        Outputs:
+        --------
+        {self.path}/{self.prototype}_fuel.xml : XML file
+            File containing the depleted material composition in the
+            required format for Cyclus. The path and prototype names are read
+            in from the class instantiation. Path will need to
+            be relative to Cyclus input file location
+        '''
+        results = od.Results.from_hdf5(str(self.path + "depletion_results.h5"))
```

Comment:
`Results.from_hdf5` is being depreciated. 
```suggestion
        results = od.Results(self.path / "depletion_results.h5")
```